src/app/src/components/FacilitiesMap.jsx

```diff
@@ -251,11 +290,119 @@ class FacilitiesMap extends Component {
         return null;
     };
 
+    displaySelectedMarker = () => {
+        const {
+            data,
+            match: {
+                params: {
+                    oarID,
+                },
+            },
+            facilityDetailsData,
+        } = this.props;
+
+        if (facilityDetailsData && (!this.markersLayer || !data)) {
+            const marker = new this.mapMethods.Marker({
+                position: new this.mapMethods.LatLng(
+                    last(facilityDetailsData.geometry.coordiantes),
+                    head(facilityDetailsData.geometry.coordinates),
+                ),
+                icon: {
+                    url: selectedMarkerURL,
+                    scaledSize: new this.mapMethods.Size(30, 40),
+                    zIndex: 1000,
+                },
+                oarID: facilityDetailsData.properties.id,
+                feature: facilityDetailsData,
+                map: this.googleMapElement,
+            });
+
+            this.detailsSelectedMarker = marker;
+        }
+
+        if (this.markersLayer && data) {
+            const markerForOARID = this.markersLayer
+                .find(marker => marker.get('oarID') === oarID);
+
+            if (markerForOARID) {
+                markerForOARID.setIcon({
+                    url: selectedMarkerURL,
+                    scaledSize: new this.mapMethods.Size(30, 40),
```

Comment:
We may want to consider extracting this literal `Size` into a constant since we repeat it several times.

Reply:
Good point, will try this out.

Reply:
I tried this out in ac98a485884393861ebe112a9fc49644544fe101 however I can't tell if it's better or not. The `Size(30, 40)` value gets set from a call to `this.mapMethods` initially, which means it gets set after the map mounts.

I used that to set a `this.scaledMarkerSize` class variable, but once I added guards to re-initialize it if necessary in places where it wasn't already initialized it ended up adding more code rather than reducing code.

Reply:
👍